src/lib/es2023.weakref.d.ts

```diff
@@ -0,0 +1,49 @@
+interface WeakKeyTypes {
+    symbol: symbol;
+}
+
+interface FinalizationRegistry<T> {
+    readonly [Symbol.toStringTag]: "FinalizationRegistry";
+
+    /**
+     * Registers an object with the registry.
+     * @param target The target object to register.
+     * @param heldValue The value to pass to the finalizer for this object. This cannot be the
+     * target object.
+     * @param unregisterToken The token to pass to the unregister method to unregister the target
+     * object. If provided (and not undefined), this must be an object. If not provided, the target
+     * cannot be unregistered.
+     */
+    registerSymbol(target: symbol, heldValue: T, unregisterToken?: symbol): void;
```

Comment:
I think instead of adding an overload here we could update the existing definition here: https://github.com/microsoft/TypeScript/blob/13c374a868c926f6a907666a5599992c1351b773/src/lib/es2021.weakref.d.ts#L35

One reason is because the `unregisterToken` doesn't need to be the same type as the `target`. i.e.:

```ts
 registerSymbol(target: symbol, heldValue: T, unregisterToken?: object): void;
 registerSymbol(target: object, heldValue: T, unregisterToken?: symbol): void;
```

Are also both valid. If we have just one definition of:

```ts
registerSymbol(target: WeakKey, heldValue: T, unregisterToken?: WeakKey): void;
```

This will cover all 4 cases. Though we will also need to update this comment https://github.com/microsoft/TypeScript/blob/13c374a868c926f6a907666a5599992c1351b773/src/lib/es2021.weakref.d.ts#L27-L33

So it doesn't refer to `object` and is more generic something like:

```js
    /**
     * Registers a value with the registry.
     * @param target The target value to register.
     * @param heldValue The value to pass to the finalizer for this target. This cannot be the
     * target value.
     * @param unregisterToken The token to pass to the unregister method to unregister the target
     * value. If not provided, the target cannot be unregistered.
     */
```